src/VisualStudio/Core/Def/Implementation/Workspace/WorkspaceFailureOutputPane.cs

```diff
@@ -49,26 +49,58 @@ private IVsOutputWindowPane OutputPane
+                    // can be made active ("selected") by the user, the latter indicates that the pane is currently active.
+                    // There's no way to only initialize a pane without also making it active so we remember the last active pane
+                    // and reactivate it after we've created ourselves to avoid stealing focus away from it.
+                    var lastActivePane = GetActivePane(outputWindow);
 
-                    // If the workspace pane has not been registered before, create it
-                    if (_doNotAccessDirectlyOutputPane == null || hr != VSConstants.S_OK)
+                    _doNotAccessDirectlyOutputPane = CreateOutputPane(outputWindow);
+
+                    if (lastActivePane != Guid.Empty)
                     {
-                        if (ErrorHandler.Failed(outputWindow.CreatePane(ref workspacePaneGuid, ServicesVSResources.IntelliSense, fInitVisible: 1, fClearWithSolution: 1)) ||
-                            ErrorHandler.Failed(outputWindow.GetPane(ref workspacePaneGuid, out _doNotAccessDirectlyOutputPane)))
-                        {
-                            return null;
-                        }
-
-                        // Must activate the workspace pane for it to show up in the output window
-                        _doNotAccessDirectlyOutputPane.Activate();
+                        ActivatePane(outputWindow, lastActivePane);
                     }
                 }
 
                 return _doNotAccessDirectlyOutputPane;
             }
         }
+
+        private IVsOutputWindowPane CreateOutputPane(IVsOutputWindow outputWindow)
+        {
+            // Try to get the workspace pane if it has already been registered
+            var workspacePaneGuid = s_workspacePaneGuid;
+
+            // If the pane has already been created, CreatePane returns it
+            if (ErrorHandler.Succeeded(outputWindow.CreatePane(ref workspacePaneGuid, ServicesVSResources.IntelliSense, fInitVisible: 1, fClearWithSolution: 1)) &&
+                ErrorHandler.Succeeded(outputWindow.GetPane(ref workspacePaneGuid, out var pane)))
+            {
+                return pane;
+            }
+
+            return null;
+        }
+
+        private static Guid GetActivePane(IVsOutputWindow outputWindow)
+        {
+            if (outputWindow is IVsOutputWindow2 outputWindow2)
+            {
+                if (ErrorHandler.Succeeded(outputWindow2.GetActivePaneGUID(out var activePaneGuid)))
+                {
+                    return activePaneGuid;
+                }
+            }
+
+            return Guid.Empty;
+        }
+
+        public static void ActivatePane(IVsOutputWindow outputWindow, Guid paneGuid)
+        {
+            if (ErrorHandler.Succeeded(outputWindow.GetPane(ref paneGuid, out var pane)))
+            {
+                pane.Activate();
```

Comment:
❔ Does this set focus to the Output Window, or just change the selection? (possible question for @olegtk)

Reply:
this eventually calls IVsWindowFrame.Show(), which makes Output window visible and sets focus to its content.

Reply:
This does not activate the Output window, see: https://devdiv.visualstudio.com/DevDiv/_git/VS?path=%2Fsrc%2Fenv%2Fmsenv%2Fcore%2Foutpwin.cpp&version=GBlab%2Fvsumain&line=2979&lineStyle=plain&lineEnd=2980&lineStartColumn=1&lineEndColumn=1. Notice how it passes `FALSE` to `setFocus`.